service/dap/server.go

```diff
@@ -132,6 +139,34 @@ func (s *Server) setLaunchAttachArgs(request dap.LaunchAttachRequest) {
 	if ok {
 		s.args.showGlobalVariables = globals
 	}
+	rules, ok := request.GetArguments()["substitutePath"]
+	if ok {
+		rulesParsed, ok := rules.([]interface{})
+		if !ok {
+			return fmt.Errorf("'substitutePath' attribute '%v' in debug configuration is not a []interface{}", rules)
```

Comment:
Unfortunately, swapping with map[string]interface or map[string]string does not work, it seems like the nesting is required.

I will change to having a single error message saying that the type is not correct instead of the more confusing intermediate steps.